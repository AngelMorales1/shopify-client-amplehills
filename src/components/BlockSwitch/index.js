import React from 'react';

import get from 'utils/get';

import ProductDetails from 'components/ProductDetails';
import ImageText from 'components/ImageText';
import ImageDoubleText from 'components/ImageDoubleText';
import ChooseYourOwnStory from 'components/ChooseYourOwnStory';
import ComicStrip from 'components/ComicStrip';
import MarkdownBlock from 'components/MarkdownBlock';
import GenericHero from 'components/GenericHero';
import HorizontalCarouselBlock from 'components/HorizontalCarouselBlock';
import EventsBlock from 'components/EventsBlock';
import ImageVideoCarousel from 'components/ImageVideoCarousel';
import FAQBlock from 'components/FAQBlock';
import ProductWhatsIncluded from 'components/ProductWhatsIncluded';
import AvailableLocations from 'components/AvailableLocations';
import PressBlock from 'components/PressBlock';
import AvailableFlavors from 'components/AvailableFlavors';
import FlavorsLanding from 'components/FlavorsLanding';
import CardsBlock from 'components/CardsBlock';

const BlockSwitch = props => {
  const { block } = props;
  const type = get(block, 'sys.contentType.sys.id');

  switch (type) {
    case 'blockProductDetails':
      return <ProductDetails {...props} />;
    case 'blockImageText':
      return <ImageText {...props} />;
    case 'blockImageDoubleText':
      return <ImageDoubleText {...props} />;
    case 'blockChooseYourOwnStory':
      return <ChooseYourOwnStory {...props} />;
    case 'blockComicStrip':
      return <ComicStrip {...props} />;
    case 'blockMarkdown':
      return <MarkdownBlock {...props} />;
    case 'blockGenericHero':
      return <GenericHero {...props} />;
    case 'blockHorizontalCarousel':
      return <HorizontalCarouselBlock {...props} />;
    case 'blockUpcomingEvents':
      return <EventsBlock {...props} />;
    case 'blockImageVideoCarousel':
      return <ImageVideoCarousel {...props} />;
    case 'blockFaq':
      return <FAQBlock {...props} />;
    case 'blockCardsModule':
      return <CardsBlock {...props} />;
    case 'blockFixedContent':
      const fields = get(block, 'fields', {});
      const contentType = get(fields, 'contentType', '');
      const drip = get(fields, 'drip', false);

      switch (contentType) {
        case "What's Included Block":
          return <ProductWhatsIncluded {...props} drip={drip} />;
        case 'Available Locations Block':
          return <AvailableLocations {...props} drip={drip} />;
        case 'Press Block':
          return <PressBlock {...props} drip={drip} />;
        case 'Available Flavors Block':
          return <AvailableFlavors {...props} drip={drip} />;
        case 'Flavors Block':
          return <FlavorsLanding {...props} drip={drip} />;
        default:
          return null;
      }
    default:
      return null;
  }
};

export default BlockSwitch;
