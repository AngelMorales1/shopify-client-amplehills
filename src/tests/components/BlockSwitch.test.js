import React from 'react';
import { shallow } from 'enzyme';

import BlockSwitch from 'components/BlockSwitch';

it('renders without data', () => {
  const component = shallow(<BlockSwitch />);

  expect(component).toMatchSnapshot();
});

it('renders with data', () => {
  const component = shallow(
    <BlockSwitch
      actions={{ foo: () => console.log('foo') }}
      addLineItemsStatus={'foo'}
      block={{
        fields: {
          backgroundColor: 'foo',
          image: {
            sys: {
              createdAt: 'foo',
              environment: { foo: 'foo' },
              id: 'foo',
              locale: 'foo',
              revision: 0,
              space: { sys: {} },
              type: 'Asset',
              updatedAt: 'foo'
            },
            fields: {
              file: {
                url: 'foo'
              },
              title: 'foo'
            }
          },
          text1: 'foo',
          text2: 'foo',
          title: 'foo',
          title1: 'foo',
          title2: 'foo'
        },
        sys: {
          contentType: { sys: {} },
          createdAt: 'foo',
          environment: { sys: {} },
          id: 'foo',
          locale: 'foo',
          revision: 0,
          space: { sys: {} },
          type: 'foo',
          updatedAt: 'foo'
        }
      }}
      checkout={{
        completedAt: null,
        currencyCode: 'foo',
        id: 'foo',
        lineItems: [{ foo: {} }],
        note: null,
        subtotalPrice: '0',
        totalPrice: '0'
      }}
      error={null}
      globalSettings={{
        facebookLink: 'foo',
        footerIllustration: { sys: {}, fields: {} },
        instagramLink: 'foo',
        ourPledgeIcon: { sys: {}, fields: {} },
        shipping2: 'foo',
        shippingDates: 'foo',
        title: 'foo',
        twitterLink: 'foo'
      }}
      history={{
        action: 'foo',
        block: () => console.log('foo'),
        createHref: () => console.log('foo'),
        go: () => console.log('foo'),
        goBack: () => console.log('foo'),
        goForward: () => console.log('foo'),
        length: 0,
        listen: () => console.log('foo'),
        location: {
          pathname: 'foo',
          search: '',
          hash: '',
          state: undefined,
          key: 'foo'
        },
        push: () => console.log('foo'),
        replace: () => console.log('foo')
      }}
      location={{
        hash: 'foo',
        key: 'foo',
        pathname: '/foo',
        search: 'foo',
        state: undefined
      }}
      match={{
        isExact: true,
        params: {
          productHandle: 'foo'
        },
        path: '/foo',
        url: '/foo'
      }}
      model={{
        action: {
          type: 'foo',
          payload: {}
        },
        isError: false,
        value: {
          sys: {},
          total: 0,
          skip: 0,
          limit: 0,
          items: []
        }
      }}
      ourPledge={{
        calloutImage: { sys: {}, fields: {} },
        overlayContentImage: { sys: {}, fields: {} },
        shippingInformation: 'foo',
        shippingPledge: 'foo',
        title: 'foo'
      }}
      ourPledgeOverlayIsOpen={false}
      product={{
        available: true,
        blocks: [{}],
        cartDetails: 'foo',
        description: 'foo',
        flavorDescription: 'foo',
        gridImage: 'foo',
        handle: 'foo',
        id: 'foo',
        pintImage: 'foo',
        price: 0,
        subItems: [],
        subItemsAvailable: true,
        title: 'foo',
        variants: [{}]
      }}
      products={{ foo: {} }}
      staticContext={undefined}
      view={() => console.log('foo')}
      z={0}
    />
  );

  expect(component).toMatchSnapshot();
});
