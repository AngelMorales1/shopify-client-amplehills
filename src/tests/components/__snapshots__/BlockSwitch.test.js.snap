// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with data 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <BlockSwitch
    actions={
        Object {
            "foo": [Function],
          }
    }
    addLineItemsStatus="foo"
    block={
        Object {
            "fields": Object {
              "backgroundColor": "foo",
              "image": Object {
                "fields": Object {
                  "file": Object {
                    "url": "foo",
                  },
                  "title": "foo",
                },
                "sys": Object {
                  "createdAt": "foo",
                  "environment": Object {
                    "foo": "foo",
                  },
                  "id": "foo",
                  "locale": "foo",
                  "revision": 0,
                  "space": Object {
                    "sys": Object {},
                  },
                  "type": "Asset",
                  "updatedAt": "foo",
                },
              },
              "text1": "foo",
              "text2": "foo",
              "title": "foo",
              "title1": "foo",
              "title2": "foo",
            },
            "sys": Object {
              "contentType": Object {
                "sys": Object {},
              },
              "createdAt": "foo",
              "environment": Object {
                "sys": Object {},
              },
              "id": "foo",
              "locale": "foo",
              "revision": 0,
              "space": Object {
                "sys": Object {},
              },
              "type": "foo",
              "updatedAt": "foo",
            },
          }
    }
    checkout={
        Object {
            "completedAt": null,
            "currencyCode": "foo",
            "id": "foo",
            "lineItems": Array [
              Object {
                "foo": Object {},
              },
            ],
            "note": null,
            "subtotalPrice": "0",
            "totalPrice": "0",
          }
    }
    error={null}
    globalSettings={
        Object {
            "facebookLink": "foo",
            "footerIllustration": Object {
              "fields": Object {},
              "sys": Object {},
            },
            "instagramLink": "foo",
            "ourPledgeIcon": Object {
              "fields": Object {},
              "sys": Object {},
            },
            "shipping2": "foo",
            "shippingDates": "foo",
            "title": "foo",
            "twitterLink": "foo",
          }
    }
    history={
        Object {
            "action": "foo",
            "block": [Function],
            "createHref": [Function],
            "go": [Function],
            "goBack": [Function],
            "goForward": [Function],
            "length": 0,
            "listen": [Function],
            "location": Object {
              "hash": "",
              "key": "foo",
              "pathname": "foo",
              "search": "",
              "state": undefined,
            },
            "push": [Function],
            "replace": [Function],
          }
    }
    location={
        Object {
            "hash": "foo",
            "key": "foo",
            "pathname": "/foo",
            "search": "foo",
            "state": undefined,
          }
    }
    match={
        Object {
            "isExact": true,
            "params": Object {
              "productHandle": "foo",
            },
            "path": "/foo",
            "url": "/foo",
          }
    }
    model={
        Object {
            "action": Object {
              "payload": Object {},
              "type": "foo",
            },
            "isError": false,
            "value": Object {
              "items": Array [],
              "limit": 0,
              "skip": 0,
              "sys": Object {},
              "total": 0,
            },
          }
    }
    ourPledge={
        Object {
            "calloutImage": Object {
              "fields": Object {},
              "sys": Object {},
            },
            "overlayContentImage": Object {
              "fields": Object {},
              "sys": Object {},
            },
            "shippingInformation": "foo",
            "shippingPledge": "foo",
            "title": "foo",
          }
    }
    ourPledgeOverlayIsOpen={false}
    product={
        Object {
            "available": true,
            "blocks": Array [
              Object {},
            ],
            "cartDetails": "foo",
            "description": "foo",
            "flavorDescription": "foo",
            "gridImage": "foo",
            "handle": "foo",
            "id": "foo",
            "pintImage": "foo",
            "price": 0,
            "subItems": Array [],
            "subItemsAvailable": true,
            "title": "foo",
            "variants": Array [
              Object {},
            ],
          }
    }
    products={
        Object {
            "foo": Object {},
          }
    }
    staticContext={undefined}
    view={[Function]}
    z={0}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): null,
  Symbol(enzyme.__nodes__): Array [
    null,
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`renders without data 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <BlockSwitch />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): null,
  Symbol(enzyme.__nodes__): Array [
    null,
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
